frontend/src/components/Forms/reports/ReportFilterForm.js

```diff
@@ -0,0 +1,216 @@
+      )[0];
+
+      dropdownMineReportCategoryOptionsFiltered = dropdownMineReportCategoryOptions.filter((cat) =>
+        selectedMineReportDefinition.categories
+          .map((category) => category.mine_report_category)
+          .includes(cat.value)
+      );
+    }
+    this.setState({
+      dropdownMineReportCategoryOptionsFiltered,
+    });
+  };
+
+  componentWillReceiveProps = (nextProps) => {
+    if (nextProps.selectedMineReportCategory !== this.props.selectedMineReportCategory) {
+      this.updateMineReportOptions(
+        nextProps.mineReportDefinitionOptions,
+        nextProps.selectedMineReportCategory
+      );
+    }
+    if (
+      nextProps.selectedMineReportDefinitionGuid !== this.props.selectedMineReportDefinitionGuid
+    ) {
+      this.updateMineReportCategoryOptions(
+        nextProps.dropdownMineReportCategoryOptions,
+        nextProps.selectedMineReportDefinitionGuid
+      );
+    }
+  };
+
+  render() {
+    return (
+      <Form
+        layout="vertical"
+        onSubmit={this.props.handleSubmit}
+        onReset={this.handleReset}
+        enableReinitialize="true"
+      >
+        <div>
+          <Row gutter={16}>
+            <Col md={6} sm={24}>
+              <Field
+                id="report_name"
+                name="report_name"
+                label="Report Name"
+                placeholder="Start typing a report name"
+                component={renderConfig.SELECT}
+                data={this.state.dropdownMineReportDefinitionOptionsFiltered}
+              />
+            </Col>
+            <Col md={6} sm={24}>
+              <Field
+                id="report_type"
+                name="report_type"
+                label="Report Type"
+                placeholder="Select a report type"
+                component={renderConfig.SELECT}
+                data={this.state.dropdownMineReportCategoryOptionsFiltered}
+              />
+            </Col>
+            <Col md={6} sm={24}>
+              <Field
+                id="compliance_year"
+                name="compliance_year"
+                label="Compliance Year"
+                placeholder="Start date"
+                component={renderConfig.YEAR}
+              />
+            </Col>
+            {/* <Col md={6} sm={24}>
+              <Field
+                id="requested_by"
+                name="requested_by"
+                label="Requested by"
+                placeholder="Start typing an inspector name"
+                component={renderConfig.SELECT}
+              />
+            </Col> */}
+          </Row>
+          <Row gutter={6}>
+            <Col lg={3} md={6} sm={24}>
+              <Field
+                id="report_due_date_start"
+                name="report_due_date_start"
+                label="Report Due Date Range"
+                placeholder="Start date"
+                component={renderConfig.DATE}
+              />
```

Comment:
@Jon has created a validate function that makes sure the start date is before the end date, could be helpful to add here once his PR is merged.